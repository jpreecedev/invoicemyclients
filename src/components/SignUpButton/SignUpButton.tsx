import React from 'react'
import { Link } from 'react-router-dom'
import firebase from 'firebase/app'
import { useAuthState } from 'react-firebase-hooks/auth'

import { ROUTES } from '../../routes'

const SignUpButton = () => {
  const [user] = useAuthState(firebase.auth())

  return (
    <>
      {!user && (
        <Link className="navbar-btn btn btn-sm btn-primary lift ml-auto" to={ROUTES.signup}>
          Sign Up
        </Link>
      )}
      {user && (
        <button
          className="navbar-btn btn btn-sm btn-primary lift ml-auto"
          onClick={() => firebase.auth().signOut()}
        >
          Log out
        </button>
      )}
    </>
  )
}

export { SignUpButton }
