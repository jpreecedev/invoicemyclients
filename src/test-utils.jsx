import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { createMemoryHistory } from 'history'

function renderWithRouter(
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  const Wrapper = ({ children }) => <Router history={history}>{children}</Router>
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  }
}

export { renderWithRouter }
