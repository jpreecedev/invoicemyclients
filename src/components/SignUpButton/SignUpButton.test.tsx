import React from 'react'

import { renderWithRouter } from '../../test-utils'
import { SignUpButton } from './SignUpButton'

describe('<SignUpButton /> tests', () => {
  it('should match snapshot', () => {
    expect(renderWithRouter(<SignUpButton />).container).toMatchSnapshot()
  })
})
