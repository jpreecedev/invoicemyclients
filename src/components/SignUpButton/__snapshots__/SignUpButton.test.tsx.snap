// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpButton /> tests should match snapshot 1`] = `
<div>
  <a
    class="navbar-btn btn btn-sm btn-primary lift ml-auto"
    href="/sign-up"
  >
    Sign Up
  </a>
</div>
`;
